docs/azure-openshift.md

```diff
@@ -24,78 +23,43 @@ $ SUBSCRIPTION_ID=$(oc get secrets azure-credentials  -n kube-system  --template
 $ TENANT_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_tenant_id}} | base64 -d)
 ```
 
-4. Login to azure with base64 decoded values you get from above
-
+4. Login to Azure with base64 decoded values you get from above:
 ```bash
 $ az login --service-principal -u "${CLIENT_ID}" -p "${CLIENT_SECRET}" --tenant "${TENANT_ID}"
 ```
 
-5. Get the routes to check the domain
+5. Get the routes to check your cluster's domain (everything after `apps`):
 ```bash
-oc get routes --all-namespaces | grep console
+$ oc get routes --all-namespaces | grep console
 openshift-console          console             console-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
 openshift-console          downloads           downloads-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
 ```
 
-6. Get the dns zone list w.r.t your resource group.
+6. Get the list of dns zones to find the one which corresponds to the previously found route’s domain:
 ```bash
 $ az network dns zone list --resource-group "${RESOURCE_GROUP}"
-[]
 ```
-Initially you will see nothing. So you will have to create a zone for your resource group.
 
-7. Create a dns zone for you resourcegroup.
+7. Create [ExternalDNS CR](https://github.com/openshift/external-dns-operator/blob/main/config/samples/azure/operator_v1alpha1_externaldns_openshift.yaml) as follows:
 ```bash
-$ az network dns zone create -g "${RESOURCE_GROUP}" -n <pick up the zone name from the route  eg- the part after apps. eg - test-azure.qe.azure.devcluster.openshift.com>
-```
-
-8. Then create [ExternalDNS CR](https://github.com/openshift/external-dns-operator/blob/main/config/samples/azure/operator_v1alpha1_externaldns_openshift.yaml) as follows -  
-```yaml
 $ cat <<EOF | oc create -f -
 apiVersion: externaldns.olm.openshift.io/v1alpha1
 kind: ExternalDNS
 metadata:
   name: sample-azure
 spec:
-  domains:
-  - filterType: Include
-    matchType: Exact
-    name: test-azure1.qe.azure.devcluster.openshift.com
+  zones:
+  - "/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure.qe.azure.devcluster.openshift.com"
   provider:
     type: Azure
   source:
+    type: OpenShiftRoute
     openshiftRouteOptions:
       routerName: default
-    fqdnTemplate:
-    - "{{.Name}}.apps.test-azure1.qe.azure.devcluster.openshift.com"
```

Comment:
Will this work as we set default hostname annotation policy ? Have you tested this ?

Reply:
Yes, this [will work](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/controller/externaldns/pod.go#L178-L184).